src/components/FieldDateInput/DateInput.js

```diff
@@ -106,12 +107,16 @@ const defaultProps = {
 
 // Checks if time slot (propTypes.timeSlot) start time equals a day (moment)
 const timeSlotEqualsDay = (timeSlot, day) => {
-  // Time slots describe available dates by providing a start and
-  // an end date which is the following day. In the single date picker
-  // the start date is used to represent available dates.
-  const localStartDate = dateFromAPIToLocalNoon(timeSlot.attributes.start);
-
-  return isSameDay(day, moment(localStartDate));
+  if (ensureTimeSlot(timeSlot).attributes.type === TIME_SLOT_DAY) {
+    // Time slots describe available dates by providing a start and
+    // an end date which is the following day. In the single date picker
+    // the start date is used to represent available dates.
+    const localStartDate = dateFromAPIToLocalNoon(timeSlot.attributes.start);
+
+    return isSameDay(day, moment(localStartDate));
+  } else {
+    return false;
+  }
```

Comment:
This works but it could be written shorter as 
```
const isDay = ensureTimeSlot(timeSlot).attributes.type === TIME_SLOT_DAY;
return isDay && isSameDay(day, moment(localStartDate))
```

Reply:
Looks elegant. One drawback though is that the `localStartDate` needs to be resolved with `dateFromAPItoLocalNoon` even though the time slot type wasn't of type day. But the `dateFromAPIToLocalNoon` should handle anything you throw at it.